contents/tutorials/mcp-analytics.mdx

```diff
@@ -0,0 +1,476 @@
+---
+title: How to set up MCP analytics and error tracking
+date: 2025-10-14
+author:
+  - arda-eren
+tags:
+  - MCP
+  - product analytics
+  - feature flags
+---
+
+MCP servers give LLMs powerful capabilities, but without analytics and error tracking you're flying blind with no visibility into usage or performance. Which tools get called? How often? Where are the bottlenecks? What's failing?
+
+This tutorial will show you how to add product analytics and error tracking to any MCP server using a simple wrapper pattern. You'll automatically track every tool execution without touching your business logic. 
+
+By the end, your MCP server will be able to:
+
+- Track execution time for every tool call
+- Capture errors with context
+- Record custom metrics
+- Sanitize sensitive data
+
+This integration pattern works for any MCP server and is vendor-agnostic. We'll use PostHog for this tutorial but you can use any developer tool you prefer.
+
+The full source code is available in this [GitHub repository](https://github.com/arda-e/mcp-posthog-analytics).
+
+## Prerequisites
+
+- Node.js 18+ 
+- PostHog account (free tier)
+- Claude Desktop or another MCP client to test your MCP server
+- Basic TypeScript knowledge
+- Code editor (e.g., VS Code, Cursor)
+
+## A high-level overview of MCP analytics
```

Comment:
I feel like you sort of already give an overview of MCP analytics before, so you can probably drop this.